atlasdb-client/build.gradle

```diff
@@ -15,8 +15,17 @@ schemas = [
 
 libsDirName = file('build/artifacts')
 
+task myTestJar(type: Jar) {
+    from sourceSets.test.output
+    classifier = 'tests'
+}
+
 configurations {
-    testArtifacts.extendsFrom testRuntime
+    testArtifacts
+}
+
+artifacts {
+    testArtifacts myTestJar
```

Comment:
Not sure I fully understand what you're asking, but...

4 projects use this: `atlasdb-cassandra`, `atlasdb-config`, `atlasdb-dbkvs`, and `atlasdb-tests-shared`
They all take it as a `testCompile` dependency like this:
```
dependencies {
    testCompile project(path: ":atlasdb-client", configuration: "testArtifacts")
}
```

I don't think this has any functional change. The set of dependencies/artifacts are still the same, but this makes it a little clearer both to the reader and gradle/IDE that `atlasdb-client`'s `testArtifacts` configuration contains the output from its test source, and other projects can depend on that.